From c6f25a8b3b7d2369c38dc400022ab15bb52d6001 Mon Sep 17 00:00:00 2001
From: Abhishek <apatgaonkar@habana.ai>
Date: Wed, 03 Nov 2021 08:40:29 +0200
Subject: [PATCH] [SW-0] gaudi2: wait on sram ext ld done in pcie link up flow

Currently, as part of pcie link up flow code waits for SRAM_INIT_DONE
and timeouts when its not set. INIT_DONE means that the PCS has finished
initializing its internal SRAM, and that always happens after
deasserting the PHY reset. Also the HW state machine itself waits for
those bits to be set. Instead of waiting on SRAM_INIT_DONE, this patch
add changes to wait on EXT_LD_DONE to be set which indicates that our
logic has finished copying the PHY FW from PIF SRAM to the Synopsys
PCS IP SRAM.

Change-Id: Ie0ec684465f95a24ac04f704adc3f7739e90e265
---

diff --git a/arch/gaudi2/common/common.c b/arch/gaudi2/common/common.c
index 3adf05d..1f7f755 100644
--- a/arch/gaudi2/common/common.c
+++ b/arch/gaudi2/common/common.c
@@ -339,32 +339,32 @@
 		regmap_writel(value, addr);
 }
 
-static void pcie_wait_for_sram_init_done(void)
+static void pcie_wait_for_sram_ext_ld_done(void)
 {
-	uint8_t sram_init_done = 0;
+	uint8_t sram_ext_ld_done = 0;
 
 	do {
-		sram_init_done =
-			regmap_readl(mmPCIE_PHY_PHYX_SRAM_INIT_DONE_0);
-	} while (sram_init_done != 1);
+		sram_ext_ld_done =
+			regmap_readl(mmPCIE_PHY_PHYX_SRAM_EXT_LD_DONE_0);
+	} while (sram_ext_ld_done != 1);
 
-	sram_init_done = 0;
+	sram_ext_ld_done = 0;
 	do {
-		sram_init_done =
-			regmap_readl(mmPCIE_PHY_PHYX_SRAM_INIT_DONE_1);
-	} while (sram_init_done != 1);
+		sram_ext_ld_done =
+			regmap_readl(mmPCIE_PHY_PHYX_SRAM_EXT_LD_DONE_1);
+	} while (sram_ext_ld_done != 1);
 
-	sram_init_done = 0;
+	sram_ext_ld_done = 0;
 	do {
-		sram_init_done =
-			regmap_readl(mmPCIE_PHY_PHYX_SRAM_INIT_DONE_2);
-	} while (sram_init_done != 1);
+		sram_ext_ld_done =
+			regmap_readl(mmPCIE_PHY_PHYX_SRAM_EXT_LD_DONE_2);
+	} while (sram_ext_ld_done != 1);
 
-	sram_init_done = 0;
+	sram_ext_ld_done = 0;
 	do {
-		sram_init_done =
-			regmap_readl(mmPCIE_PHY_PHYX_SRAM_INIT_DONE_3);
-	} while (sram_init_done != 1);
+		sram_ext_ld_done =
+			regmap_readl(mmPCIE_PHY_PHYX_SRAM_EXT_LD_DONE_3);
+	} while (sram_ext_ld_done != 1);
 }
 
 static void pcie_phy_set_sram_ext_ld_done(void)
@@ -689,8 +689,8 @@
 	/* Disable the EQ REDO */
 	regmap_writel(0x00402801, mmPCIE_DBI_GEN3_RELATED_OFF);
 
-	/* Wait for init_done before executing W/A */
-	pcie_wait_for_sram_init_done();
+	/* Wait for ext_ld_done before executing W/A */
+	pcie_wait_for_sram_ext_ld_done();
 
 	pcie_phy_set_sram_ext_ld_done();
 
